internal/fnruntime/container.go

```diff
@@ -250,3 +194,48 @@ func (e *ContainerImageError) Error() string {
 		"Error: Function image %q doesn't exist remotely. If you are developing new functions locally, you can choose to set the image pull policy to ifNotPresent or never.\n%v",
 		e.Image, e.Output)
 }
+
+// filterDockerCLIOutput filters out docker CLI messages
+// from the given buffer.
+func filterDockerCLIOutput(in io.Reader) string {
```

Comment:
What is the motivation for filtering the docker output from the buffer?

I haven't encountered this use case before, but I expected there would be a more elegant way to do this (e.g docker cli flag, I/O redirection, etc)

Reply:
so kpt invokes the functions using cmd.Exec() the tricky thing is the output stream (stdout/stderr) is shared between the docker command as well as the function itself. And at the kpt level, we want to process these outputs differently. I did investigate if docker provides a simple mechanism to control its progress output, but unfortunately couldn't find one, so had to settle for this hack.

I thought a lot about the maintenance aspect of this hack. docker is a very stable CLI, so I don't expect a huge churn in this code and also, this filtering is only relevant in case of error so even if the filtering is not perfect, it isn't going to effect the user experience much. And the upside of removing the two commands (`docker pull` and `docker image inspect`)  outweighs the downside of this hack, so went with this.

Reply:
Seems reasonable to me 🙂